File: algorithms/2_medium/M/03987_minimum_total_cost_to_process_all_elements/solution.c
#define MOD 1000000007

int minimumCost(const int* nums, const int len, const int k)
{
    int64_t sum = 0;
    for (int i = 0; i < len; i += 1) {
        sum += nums[i];
    }
    const int64_t ops = (sum - 1) / k;
    const int64_t costs = (((ops % MOD) * ((ops + 1) % MOD)) / 2) % MOD;
    return (int)costs;
}

File: algorithms/2_medium/M/03987_minimum_total_cost_to_process_all_elements/solution.rs
impl Solution {
    const MOD: i64 = 1_000_000_007;

    pub fn minimum_cost(nums: Vec<i32>, k: i32) -> i32 {
        let k = k as i64;
        let mut sum = nums.iter().map(|x| *x as i64).sum::<i64>();
        let ops = (sum - 1) / k;
        ((ops % Self::MOD) * ((ops + 1) % Self::MOD) / 2 % Self::MOD) as i32
    }
}
